### PR TITLE
BGDIINF_SB-2765: configure keep-alive in socket

### PR DESCRIPTION
**Pull request by ltclm**

### CODE REVIEW

**Review by hansmannj — APPROVED** [review]
LGTM, hope it helps :+1:

Probably a short comment in the code or in the PR description/commit message might help our future selfs to remember, why we did those changes?

File: app/helpers/s3.py [hunk 1]
```diff
@@ -62,6 +63,14 @@ def get_s3_file(wmts_path, etag=None):
...
         )
         http_client.request("GET", path, headers=headers)
+        my_socket = http_client.sock
+        # Set the following socket options
+        # SO_KEEPALIVE: 1 => Enable TCP keepalive
+        # TCP_KEEPIDLE: 60 => Time in seconds until the first keepalive is sent
+        # TCP_KEEPINTVL: 60 => How often should the keepalive packet be sent
+        my_socket.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
+        my_socket.setsockopt(socket.SOL_TCP, socket.TCP_KEEPIDLE, 120)
+        my_socket.setsockopt(socket.SOL_TCP, socket.TCP_KEEPINTVL, 120)
```
Comment (ltshb) [inline app/helpers/s3.py:73]: Should we not set the socket option before sending the request on line 65 ? Based on docu `http.client.HTTPConnection.request()` send the request.

**Review by ltshb — CHANGES_REQUESTED** [review]
I'm not sure that will work, see my comment

File: app/helpers/s3.py (hunk 1 above)
Comment (ltclm) [reply app/helpers/s3.py:73]: thank you @ltshb 
yes i was not aware of that. i will try to fix this.

**Review by ltshb — APPROVED** [review]
Great thanks

**Pull request merged**